db/migrate/20230728104825_add_index_to_checklist_items.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class
+ AddIndexToChecklistItems < ActiveRecord::Migration[7.0]
+  def change
+    add_index :checklist_items, :position
```

Comment:
it should be unique for each checklist

Reply:
The database uniqueness constraint for the checklist (checklist_id) is already added to the checklist table.
```
  create_table "checklist_items", force: :cascade do |t|
    t.string "text", null: false
    t.boolean "checked", default: false, null: false
    t.bigint "checklist_id", null: false
    t.datetime "created_at", precision: nil, null: false
    t.datetime "updated_at", precision: nil, null: false
    t.bigint "created_by_id"
    t.bigint "last_modified_by_id"
    t.integer "position"
    t.index "trim_html_tags((text)::text) gin_trgm_ops", name: "index_checklist_items_on_text", using: :gin
    t.index ["checklist_id"], name: "index_checklist_items_on_checklist_id"
    t.index ["created_by_id"], name: "index_checklist_items_on_created_by_id"
    t.index ["last_modified_by_id"], name: "index_checklist_items_on_last_modified_by_id"
  end
  ```

Reply:
Where? I don't see it

Reply:
How about this
    `t.index ["checklist_id"], name: "index_checklist_items_on_checklist_id"`

Reply:
should be something like this:

t.index ["checklist_id", "position"], name: "index_checklist_items_on_checklist_id_and_position", unique: true

Reply:
The unique: true will make it impossible to reorder item positions as it would raise an error (uniqueness) when reassigning positions during reordering. This could occur when reordering positions or deleting an item up the list that will trigger the rest to be reordered.